example_code/assertj/src/test/java/hamcrest/HamcrestNullAssertions.java

```diff
@@ -0,0 +1,154 @@
+
+            assertThat(nullValue, org.hamcrest.Matchers.is(org.hamcrest.CoreMatchers.nullValue()));
+            assertThat(nullValue, org.hamcrest.Matchers.is(org.hamcrest.CoreMatchers.nullValue(Object.class)));
+            assertThat(nullValue, org.hamcrest.Matchers.is(org.hamcrest.core.IsNull.nullValue()));
+            assertThat(nullValue, org.hamcrest.Matchers.is(org.hamcrest.core.IsNull.nullValue(Object.class)));
+            assertThat(nullValue, org.hamcrest.Matchers.is(org.hamcrest.Matchers.nullValue()));
+            assertThat(nullValue, org.hamcrest.Matchers.is(org.hamcrest.Matchers.nullValue(Object.class)));
+        }
+
+        @Test
+        void test_isNullValue_with_reason() {
+            assertThat("reason", nullValue, org.hamcrest.CoreMatchers.is(org.hamcrest.CoreMatchers.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.CoreMatchers.is(org.hamcrest.CoreMatchers.nullValue(Object.class)));
+            assertThat("reason", nullValue, org.hamcrest.CoreMatchers.is(org.hamcrest.core.IsNull.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.CoreMatchers.is(org.hamcrest.core.IsNull.nullValue(Object.class)));
+            assertThat("reason", nullValue, org.hamcrest.CoreMatchers.is(org.hamcrest.Matchers.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.CoreMatchers.is(org.hamcrest.Matchers.nullValue(Object.class)));
+
+            assertThat("reason", nullValue, org.hamcrest.core.Is.is(org.hamcrest.CoreMatchers.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.core.Is.is(org.hamcrest.CoreMatchers.nullValue(Object.class)));
+            assertThat("reason", nullValue, org.hamcrest.core.Is.is(org.hamcrest.core.IsNull.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.core.Is.is(org.hamcrest.core.IsNull.nullValue(Object.class)));
+            assertThat("reason", nullValue, org.hamcrest.core.Is.is(org.hamcrest.Matchers.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.core.Is.is(org.hamcrest.Matchers.nullValue(Object.class)));
+
+            assertThat("reason", nullValue, org.hamcrest.Matchers.is(org.hamcrest.CoreMatchers.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.Matchers.is(org.hamcrest.CoreMatchers.nullValue(Object.class)));
+            assertThat("reason", nullValue, org.hamcrest.Matchers.is(org.hamcrest.core.IsNull.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.Matchers.is(org.hamcrest.core.IsNull.nullValue(Object.class)));
+            assertThat("reason", nullValue, org.hamcrest.Matchers.is(org.hamcrest.Matchers.nullValue()));
+            assertThat("reason", nullValue, org.hamcrest.Matchers.is(org.hamcrest.Matchers.nullValue(Object.class)));
+        }
+    }
+
+    @Nested
+    class NotNull {
+        private final Object notNullValue = new Object();
+
+        @Test
+        void test_nullValue() {
```

Comment:
minor: I think the test names should mention notNull instead of null
```suggestion
        void test_notNullValue() {
```